### PR TITLE
Automatically figure out where to write GIT_COMMIT

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
In most cases, we will want to put GIT_COMMIT in a directory with the same name as the project (with hyphens replaced by underscores.) Put GIT_COMMIT there by default. Also rename _edit_git_commit() to _write_git_commit(), and don't create a backup file with the old GIT_COMMIT.

### CODE REVIEW

File: rapids_build_backend/impls.py
Comment (jameslamb) [inline rapids_build_backend/impls.py:121]:
```suggestion
def _write_git_commit(config, project_name: str):
```

File: rapids_build_backend/impls.py
Comment (jameslamb) [inline rapids_build_backend/impls.py:133]:
```suggestion
    if commit is not None:
```

Since `commit_file`'s default is `""`, it being `None` at this point in the code would be a bug, wouldn't it?  If so, I think we should remove this `commit_file is not None` condition, to limit the risk of accidentally, silently skipping writing the commit file.

**Review by jameslamb — APPROVED** [review]
Thanks! I agree with the renaming. Changes and tests all look good to me.

I left 2 small suggestions for your consideration, but approving so if you agree to both of them you don't need to wait on another review from me.

File: rapids_build_backend/impls.py
Comment (jameslamb) [reply rapids_build_backend/impls.py:133]: ah interesting, got it!

Well yeah if it's not possible (https://github.com/toml-lang/toml/issues/30) then I think this condition should be removed.

File: README.md
Comment (vyasr) [inline README.md:21]: Probably OK for now, but one consequence of this changes it that it is no longer for a project to state that it does not want a commit file embedded. If we change the behavior in this way is there any way we could reenable that in the future without a breaking change? Just trying to future-proof the design.

File: rapids_build_backend/impls.py
Comment (vyasr) [inline rapids_build_backend/impls.py:121]: Can we document (briefly) how `project_name` is used and why it's relevant for writing a git commit?

File: tests/test_packages.py
Comment (vyasr) [inline tests/test_packages.py:70]: Why did this directory (and the one below) change?

File: README.md
Comment (KyleFromNVIDIA) [reply README.md:21]: See my discussion at https://github.com/rapidsai/rapids-build-backend/pull/30#discussion_r1595922544. I had initially thought that I could pass a `null` value to signify this, but it seems that toml doesn't support `null`.

If we want to allow disabling it entirely then we have a tristate: off, auto, and on (with a specified path.) I'll think about ways to express this.

File: tests/test_packages.py
Comment (KyleFromNVIDIA) [reply tests/test_packages.py:70]: This is where RBB puts the `GIT_COMMIT` file, and the directory has to exist, otherwise it throws a file not found error.

File: README.md
Comment (KyleFromNVIDIA) [reply README.md:21]: What if we allowed zero-or-more commit files? So the following are all valid:

```toml
commit-files = [
    "package/GIT_COMMIT",
]
```

```toml
commit-files = [
    "package_a/GIT_COMMIT",
    "package_b/GIT_COMMIT",
]
```

```toml
commit-files = []  # No files are written
```

And if `commit-files` isn't specified then it does the automatic behavior that we normally want.

Comment (jameslamb) [reply README.md:21]: oh that's a really nice way to solve this. And then if `[]` is passed, `rapids-build-backend` doesn't even both trying to determine the commit?

I support that!

Comment (KyleFromNVIDIA) [reply README.md]: I've implemented this idea.

Comment (vyasr) [reply README.md]: Sorry I'm slow in responding, but yes that seems fine to me.

**Review by vyasr — APPROVED** [review]

**Pull request merged**
